openbb_terminal/portfolio/portfolio_engine.py

```diff
@@ -284,6 +285,19 @@ def __preprocess_transactions(self):
             self.__transactions["Quantity"] = (
                 abs(self.__transactions["Quantity"]) * self.__transactions["Signal"]
             )
+            
+            # Adjust quantity and price for splits 
+            for ticker in self.__transactions["Ticker"].unique() :
+                try :
+                    splits_df = get_splits(ticker)
+                    if not splits_df.empty :
+                        splits_df = splits_df.tz_localize(tz=None)
+                        for split_date in splits_df.index :
+                            self.__transactions['Quantity'] = np.where((self.__transactions['Ticker'] == ticker) & (self.__transactions['Date'] < split_date), self.__transactions['Quantity'] * splits_df.loc[split_date].values , self.__transactions['Quantity'])
```

Comment:
I think you have the condition reversed.  If the date is > split_date, then we multiply.  For np.where it yields the first value if condition is True.

We also need to be careful and make sure that get_splits is sliced so that anything that happened before our initial purchase doesn't count )i.e, `splits_df = splits_df[splits_df.index > self.__transactions['Date'][0]]`

Reply:
If we are holding the scrip before the event (split)  then quantity will be multiplied. I think that the condition is right. I request you to run with an example and check. AAPL for example 1 share bought before 1987-06-16 should be 224 shares now. 

Slicing is also taken cared off. AAPL shares bought during 2010 should be 28 shared today.

Reply:
So I have an example.  If I bought 1 share of appl at the beginning of 2005, it underwent splits of 2,7,4, so 1 share then = 56 now.  Using your code says we initially bought 56:
![Screenshot 2023-02-10 at 8 30 14 AM](https://user-images.githubusercontent.com/18151143/218104075-29a66fc1-4031-4f25-88f6-88c64a446893.png)

If I switch the < to > but dont slice split, it accounts for everything historical, saying I started at 4 shares
![Screenshot 2023-02-10 at 8 29 13 AM](https://user-images.githubusercontent.com/18151143/218103930-aedadb35-53e1-432b-8fc3-91969a375705.png)

If I slice splits to not include before we bought, we get the desired behavior:

![Screenshot 2023-02-10 at 8 31 49 AM](https://user-images.githubusercontent.com/18151143/218104352-ab7365d5-df68-43f5-9843-4c1127c959e8.png)

Reply:
I think I see where I am looking differently.  Your method is correct if I enter into the template how many shares I bought from adjusted price.  So if  I bought 400$ of appl a few years ago, I only got 1 share.  If you enter that as 4 shares at 100$ into your excel, then your split adjustment is correct.  If you enter into the excel as 1 share at 400, then you need to do the reverse way I have here.

Reply:
In the PR I have inserted the code in the portfolio_engine.py under the preprocess_transactions..  I modify the quantity immediately after reading the transactions so that rest is taken cared of. I checked with GOOG and AAPL and it worked correctly.

Reply:
I see whats conflicting here. You are applying to the data frame that contains historical price data. Im applying the logic to transactions itself ( before it gets the historical price). I found it easier to adjust price and quantity in transactions itself.